src/Database/LSMTree/Internal/Run/Index/Compact.hs

```diff
@@ -426,11 +427,11 @@ import           Database.LSMTree.Internal.Serialise
 data CompactIndex = CompactIndex {
     -- | \(P\): Maps a page @i@ to the 32-bit slice of primary bits of its
     -- minimum key.
-    ciPrimary              :: !(VU.Vector Word32)
+    ciPrimary              :: !(VP.Vector Word32)
     -- | \(RF\): A vector that partitions 'ciPrimary' into sub-vectors
     -- containing elements
     -- that all share the same range-finder bits.
-  , ciRangeFinder          :: !(VU.Vector Word32)
+  , ciRangeFinder          :: !(VP.Vector Word32)
```

Comment:
IIRC you can access the underlying primitive vectors when using unboxed vectors, so maybe it's not "technically" necessary to change these to primary vectors because they are already primary vectors under the hood?

Reply:
The constructor for `Vector Word32` is not exported from the module we import, but I just realised I can get it from `Data.Vector.Unboxed.Base`. Nice.

Reply:
So one argument for going with primitive vectors is that we can allocate them ourselves as pinned, which will make later I/O cheaper (no copying needed).

Reply:
We can still allocate them ourselves, since they're just a newtype wrapper around the primitive ones. Of course at some point it might make sense to make it more obvious that we are relying on them internally being primitive vectors and just use primitive vectors directly.